File: PregnancyAppFrontend/src/app/dtos/weekly-survey/weekly-survey-dto.ts
import {BloodPressure} from "../../enums/weekly-survey/blood-pressure";
import {WaterConsumed} from "../../enums/weekly-survey/water-consumed";
import {Stool} from "../../enums/weekly-survey/stool";
import {Urination} from "../../enums/weekly-survey/urination";

export interface WeeklySurveyDto {
  hasOrvi: boolean;
  orviSymptoms?: string;
  hasUnordinaryTemp: boolean;
  unordinaryTempOccurrences?: number;
  unordinaryBloodPressure: BloodPressure;
  hasGynecologicalSymptoms: boolean;
  gynecologicalSymptoms?: string;
  hasUnordinaryUrine: boolean;
  hasSwelling: boolean;
  swellingDescription?: string;
  waterConsumed: WaterConsumed;
  stool: Stool;
  urination: Urination;
  weightAdded: number;
  pregnancyWeek: number;
  creationDateUtc: string;
  id: string;
}

File: PregnancyAppBackend/Dtos/Web/WeeklySurvey/WeeklySurveyDto.cs
using PregnancyAppBackend.Enums.WeeklySurvey;

namespace PregnancyAppBackend.Dtos.Web.WeeklySurvey;

public class WeeklySurveyDto
{
    public Guid Id { get; set; }
    public bool HasOrvi { get; set; }
    public string? OrviSymptoms { get; set; }
    public bool HasUnordinaryTemp { get; set; }
    public int? UnordinaryTempOccurrences { get; set; }
    public BloodPressure UnordinaryBloodPressure { get; set; }
    public bool HasGynecologicalSymptoms { get; set; }
    public string? GynecologicalSymptoms { get; set; }
    public bool HasUnordinaryUrine { get; set; }
    public bool HasSwelling { get; set; }
    public string? SwellingDescription { get; set; }
    public WaterConsumed WaterConsumed { get; set; }
    public Stool Stool { get; set; }
    public Urination Urination { get; set; }
    public decimal WeightAdded { get; set; }
    public int PregnancyWeek { get; set; }
    public DateTime? CreationDateUtc { get; set; }
}

